test/avocado/testlib_avocado/libdisc.py

```diff
@@ -22,74 +22,75 @@
 
     def createparttable(self, name, parttable='msdos'):
-        process.run("parted /var/tmp/%s_%s.target mktable %s" % (self.domain, name, parttable), shell=True)
+        self.machine.execute("sudo parted /var/tmp/%s_%s.target mktable %s" % (self.domain, name, parttable))
 
     def adddisc(self, targetsuffix):
         self.addtarget(targetsuffix)
-        process.run("iscsiadm -m discovery -t sendtargets -p %s" % self.ip, shell=True)
-        process.run("iscsiadm -m node --targetname=%s.%s:%s --login" % (self.prefix, self.domain, targetsuffix), shell=True)
-        tmp = process.run("iscsiadm -m node", shell=True)
-        print(tmp.stdout.decode("utf-8"))
-        tmp = process.run("sleep 5; iscsiadm -m session -P 3 | tail -1", shell=True)
-        tmp1 = re.search(r'Attached scsi disk\s+([a-z]*)\s+', tmp.stdout.decode("utf-8"))
+        self.machine.execute("sudo iscsiadm -m discovery -t sendtargets -p %s" % self.ip)
+        self.machine.execute("sudo iscsiadm -m node --targetname=%s.%s:%s --login" % (self.prefix, self.domain, targetsuffix))
+        tmp = self.machine.execute("sudo iscsiadm -m node")
+        print(tmp("utf-8"))
+        tmp = self.machine.execute("sleep 5; sudo iscsiadm -m session -P 3 | tail -1")
+        tmp1 = re.search(r'Attached scsi disk\s+([a-z]*)\s+', tmp("utf-8"))
         return "/dev/%s" % str(tmp1.group(1))
 
     def deldisc(self, name):
         targetname = "%s.%s:%s" % (self.prefix, self.domain, name)
-        process.run("iscsiadm -m node -T %s --portal %s -u" % (targetname, self.ip), shell=True)
+        self.machine.execute("sudo iscsiadm -m node -T %s --portal %s -u" % (targetname, self.ip))
         self.targetlist.remove(name)
 
     def clear(self):
         for foo in self.targetlist:
             self.deldisc(foo)
-        process.run("iscsiadm -m discovery -p %s -o delete" % self.ip, shell=True)
-        process.run("targetcli clearconfig confirm=True", shell=True)
-        process.run("targetcli saveconfig", shell=True)
+        self.machine.execute("sudo iscsiadm -m discovery -p %s -o delete" % self.ip)
+        self.machine.execute("sudo targetcli clearconfig confirm=True")
+        self.machine.execute("sudo targetcli saveconfig")
 
 
 class DiscSimple():
-    def __init__(self, location='/var/tmp'):
+    def __init__(self, machine, location='/var/tmp'):
+        self.machine = machine
         self.location = location
         self.targetlist = []
 
     def adddisc(self, name, size=999):
         outfile = "%s/%s" % (self.location, name)
-        process.run("dd if=/dev/zero of=%s bs=1M count=1 seek=%d" % (outfile, size), shell=True)
-        process.run("sudo losetup -f %s; sleep 1" % outfile, shell=True)
-        tmp = process.run("sudo losetup -j %s" % outfile, shell=True)
-        tmp1 = re.search(r'(/dev/loop[0-9]+):\s+', tmp.stdout.decode("utf-8"))
+        self.machine.execute("sudo dd if=/dev/zero of=%s bs=1M count=1 seek=%d" % (outfile, size))
```

Comment:
I am confused, why process.run doesn't need sudo but machine.execute does?

Reply:
This is little bit tricky:
by default ``machine.execute`` uses ``root``, for virtual machines.
also by default tests were scheduled under root, so that ``process.run`` used root credentials.
But in selenium lib, there is used ``test`` user what is inside sudoers users, so that sudo is necessary. and then ``machine.execute`` commands are done under test user.
and also when you run ``sudo command`` under root, it is fine, so that this ensure that process have root privileges.